scorecard.js

```diff
@@ -0,0 +1,21 @@
+const Frame = require('./frame');
+
+class Scorecard {
+  constructor() {
+    this.frames = []
+  }
+
+  inputFrame(rolls) { //rolls will be e.g. [2,3]
+    const frame = new Frame(rolls); // would it be better to inject this in? If I inject it, would I then need to use a double in the tests?
+    this.frames.push(frame);
+  }
+
+  score() {
+    let total = 0; 
+    this.frames.map((frame, index, frames) => {
+      total += frame.score(frames[index+1], frames[index+2])
+    });
+    return total; 
```

Comment:
This is so close to being perfectly functional ;)
Genberally, we avoid using map with functions that have side effects (meaning they change the state around them), in this case changing the total.
You can use a foreach instead, which typically is used with functions that have side effects.
Or, you could go full functional style and use a reduce here. (functional style hates side effects).
Doc for reduce - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce

Let me know if you want help figuring out how to write a reducer here.